main/src/ca/uwaterloo/flix/language/phase/njvm/Mnemonics.scala

```diff
@@ -89,4 +150,118 @@ object Mnemonics {
+  }
+
+  class LocalC[T](val localType: JvmType, val location: Int) extends Local[T]
+
+  /**
+    * Capability which allows to get/put a field
+    */
+  sealed trait Field[T]{
+    val fieldName : String
+    val fieldType : JvmType
+
+    def GET_FIELD[R <: Stack]() : F[R ** JvmType.Reference] => F[R ** T] =  t => t.getField(fieldName, fieldType)
+    def PUT_FIELD[R <: Stack]() : F[R ** JvmType.Reference ** T] => F[R] =  t => t.putField(fieldName, fieldType)
+  }
+
+  class FieldC[T](val fieldName: String, val fieldType: JvmType) extends Field[T]
+
+  sealed trait FunSig[R] {
+    val modifiers : List[JvmModifier]
+    val methodName : String
+    val returnType : JvmType
+
+    def getArgsType : List[JvmType]
+  }
+
+  class FunSig0[R](val modifiers : List[JvmModifier], val methodName : String, val returnType : JvmType,
+                   classType: JvmType) extends FunSig[R] {
+   val arg0 : Local[JvmType.Reference] = new LocalC[JvmType.Reference](classType, 0)
+
+    override def getArgsType: List[JvmType] = List()
+  }
+
+  class FunSig1[T1, R](val modifiers : List[JvmModifier], val methodName : String, val returnType : JvmType,
+                       classType: JvmType, arg1Type: JvmType) extends FunSig[R] {
+    val arg0 : Local[JvmType.Reference] = new LocalC[JvmType.Reference](classType, 0)
+    val arg1 : Local[T1] = new LocalC[T1](arg1Type, 1)
+
+    override def getArgsType: List[JvmType] = List(arg1Type)
+  }
+
+
+  trait MethodHandler0Void {
+    val invokeCode : JvmModifier
+    val className : String
+    val methodName: String
+    val args : List[JvmType]
+    val returnType : JvmType
+
+    def invoke[R <: Stack ](): F[R ** JvmType.Reference] => F[R] =
+      t => t.invoke(invokeCode, className, methodName, args, returnType)
+  }
+
+  class MethodHandler0VoidC(val invokeCode : JvmModifier, val  className : String, val methodName: String,
+                            val args : List[JvmType], val returnType : JvmType) extends MethodHandler0Void
+
+
+  def mkMethodHandler0Void[R <: Stack](invokeCode : JvmModifier, className : String, methodName: String,
+                                       args : List[JvmType], returnType : JvmType) : MethodHandler0Void=
+  { new MethodHandler0VoidC(invokeCode,className, methodName, args, returnType) }
+
+
+  class ClassGenerator(classType: JvmType.Reference,
+                       modifiers : List[JvmModifier], superClass : String, implementedInterfaces : Array[String])
+                      (implicit flix: Flix) {
+
+    private val cw: ClassWriter = {
+      val cw = AsmOps.mkClassWriter()
+      val modifierVal = modifiers.map(modifier => modifier.toInternal).sum
+      cw.visit(AsmOps.JavaVersion, modifierVal, classType.name.toInternalName, null,
+        superClass, implementedInterfaces)
+      cw
+    }
+
+    def GenMethod[R](funSig : FunSig[R], frameTransformer : F[StackNil] => F[StackNil]) : Unit = {
```

Comment:
Scala methods are always lowercase.

Reply:
I would like if we could come up with some short name for function signatures, e.g. `f` or `s`. Similarly for `frameTransformer`.